backend/src/session/Session.ts

```diff
@@ -138,14 +158,19 @@ export class SessionData {
 export function session(db: DB, logger: Logger): RequestHandler {
     return (req, res, next) => {
         req.session = new Session(db, logger, req, res);
-        req.session.restore()
-            .then(() => {
+        return (async () => {
```

Comment:
this is really hard to read - too much async calls.
Are you sure you need to return async function from current function? It wasn't returning it previously.
If `session` can return async function this could be simplified a lot by throwing away return statement

Reply:
Its not returning an async function — its returning a promise — a result of async function call.

Previously it was using promis API - then/catch, but the async logic got expanded with more async calls, so I rewrote it by introducing an async block `async () => {...}` which is immediately called (returning the promise).

I understand the readability concern, probably worth adding a comment, but I don't think that alternative (e.g. using promise API, as before) would be more readable.